File: demo/ChengFenStore/Data/OrderRepository.cs
using ChengFenStore.Models;
using System.Collections.Generic;
using System.Linq;
using Microsoft.EntityFrameworkCore;

namespace ChengFenStore.Data
{
    public interface IOrderRepository
    {
        void AddOrder(Order order);
        Order GetOrderById(int orderId);
        void UpdateOrder(Order order);
    }

    public class OrderRepository : IOrderRepository
    {
        private readonly AppDbContext _context;

        public OrderRepository(AppDbContext context)
        {
            _context = context;
        }

        public void AddOrder(Order order)
        {
            _context.Orders.Add(order);
            _context.SaveChanges();
        }

        public Order GetOrderById(int orderId)
        {
            return _context.Orders.FirstOrDefault(o => o.OrderId == orderId);
        }

        public void UpdateOrder(Order order)
        {
            _context.Orders.Update(order);
            _context.SaveChanges();
        }
    }
}

File: demo/ChengFenStore/data/order_repository.py
from .db import db
from ..models.order import Order

class OrderRepository:
    def add_order(self, order):
        db.session.add(order)
        db.session.commit()

    def get_order_by_id(self, order_id):
        return Order.query.get(order_id)

    def update_order(self, order):
        db.session.commit()
